skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt

```diff
@@ -57,15 +58,19 @@ class SkaffoldConfigurationDetector(val project: Project) : ProjectComponent {
     )
 
     override fun projectOpened() {
-        val skaffoldFiles: List<VirtualFile> =
-            SkaffoldFileService.instance.findSkaffoldFiles(project)
-        if (skaffoldFiles.isNotEmpty()) {
-            val skaffoldRunConfigList: List<RunConfiguration> =
-                getRunManager(project)
-                    .allConfigurationsList.filter { it is AbstractSkaffoldRunConfiguration }
-            if (skaffoldRunConfigList.isEmpty()) {
-                // existing Skaffold config files, but no skaffold configurations, prompt
-                showPromptForSkaffoldConfigurations(project, skaffoldFiles[0])
+        // on project open there may be a lot of indexing and other preparation and files still
+        // not available, wait for "smart" mode for configuration identification
+        DumbService.getInstance(project).runWhenSmart {
```

Comment:
nice! I recall doing something similar for services that didn't need to wait for any indexing at all: you'd simply implement the `DumbAware' marker interface. I wonder if there is something like this for a  "smart" service that we could use from a ProjectComponent

Reply:
You're right, `DumbAware` can do its action when indexing is on in background! I guess it's more useful for menu items or toolbar actions so a user can see what is still available.

Reply:
Right. I was just wondering for this PR if there was a marker interface that ProjectComponent could implement (much in the same way that we do for `DumbAware` actions) instead of having to wrap the code in `runWhenSmart`. I guess not though - I didn't see anything when quickly looking.

Reply:
Agree, I looked initially as well and didn't see anything looking useful yet.